### ISSUE TITLE
Constructor property promotion and backslash as a namespace operator cause false positive for UndefinedVariable bug

### ISSUE THREAD

**Issue by yokozawa0701** [sirbrillig/phpcs-variable-analysis#331]
Here's a simple example that generates UndefinedVariable for using Constructor property promotion and backslash as a namespace operator.

```php
class Test
{
    public function __construct(
        public \App\Models\User $user,  // Unused function parameter $user. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
    ) {}
}
```

```php
use App\Models\User;

class Test
{
    public function __construct(
        public User $user,  // Thid is no error.
    ) {}
}
```

I am using v2.11.19.
Thank you.

**Comment by sirbrillig** [comment]
Thanks for the report! Should be fixed by https://github.com/sirbrillig/phpcs-variable-analysis/pull/333

**Comment by yokozawa0701** [comment]
Thank you so much!
